NETCORE/src/Shared/Extensions/ApplicationInsightsServiceOptions.cs

```diff
@@ -116,11 +116,25 @@ public class ApplicationInsightsServiceOptions
         public bool EnableAuthenticationTrackingJavaScript { get; set; } = false;
 #endif
 
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.DiagnosticsTelemetryModule"/> should be enabled.
+        /// </summary>
+        public bool EnableDiagnosticsTelemetryModule { get; set; } = true;
```

Comment:
there is a `EnableHeartbeat` field in this class. What would happen to it. Do we allow heartbeat to be turned on, while turning off diagnostictelemetrymodule? or diagnosticmodule must be turned on as a pre-req for heartbeat?

Reply:
I think DiagnosticModule should still be allowed to turn on when Heartbeat is turned off.

Reply:
I think DiagnosticModule should still be allowed to turn on when Heartbeat is turned off. - this is correct.

My question was other way around - If we disable DiagnosticModule, should HeartBeat be still allowed?

Reply:
Gotcha. I took a look at the implementation and there's not a straight forward way to Disable the internal EventListener.

https://github.com/microsoft/ApplicationInsights-dotnet/blob/0fd652cc2bdc98ce54bda0b293c18af194a1b7a0/BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs#L34

I could introduce a new boolean `IsDiagnosticsEnable` and then conditionally set the EventListener to NULL. 
I think this is doable, but I don't know what other side effects this may cause. 
I would ask that we consider this an enhancement and postpone this specific change.

What do you think?

Reply:
why not just dispose it if not enabled?

Reply:
Should I change `options.EnableDiagnosticsTelemetryModule` to `options.EnableDiagnostics`?
I would also introduce a new property `DiagnosticsTelemetryModule.IsDiagnosticsEnabled`.

This would match the implementation of `options.EnableHeartbeat` and `DiagnosticsTelemetryModule.IsHeartbeatEnabled`.

Or would you expect that users could also remove the `DiagnosticsTelemetryModule` completely?